### DISCUSSION TITLE
Feature request: TTF file that uses CPAL/COLR tables for duotone rather than separate glyphs

### DISCUSSION THREAD

**Discussion by englercj** [FortAwesome/Font-Awesome#20925] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

My font rendering environments do not have any easy way to render stacked glyphs. I'm rendering in native code by rasterizing the TTF files to a texture using FreeType. I then render rects of that texture for glyphs in text.

**Describe the solution you'd like**

It would be great if there was a exported TTF that used the CPAL/COLR tables of the font file to do the primary/secondary coloring and used layers rather than separating them into different glyphs. That would allow me to easily rasterize each individual glyphs by using the built-in font color layer handling.

**Describe alternatives you've considered**

I'd rather not rewrite my text layout engine to support stacked glyphs right now, especially if it isn't too hard to export a font file with the standard layer tables.

**Additional context**

Let me know if there is anything else you need.

**Feature request checklist**
- [x] This is a single feature (i.e. not a re-write of all of Font Awesome)
- [x] The title starts with "Feature request: " and is followed by a clear feature name (Ex: `Feature request: moar cowbell`)
- [x] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome-Pro/issues) and to the best of my knowledge this is not a duplicate

**Comment by GreyXor** [comment]
I have the same need, in the OTF duotone version. that would make this https://gitlab.gnome.org/GNOME/pango/-/work_items/886#note_2751953 possible
<img width="817" height="200" alt="image" src="https://github.com/user-attachments/assets/09346539-2d2a-4d3a-b677-40ca1c5e6498" />
directly in pango markup
with COLR table for in OTF

Maybe we could extract glyph and create a new OTF font with COLR/CPAL, from separate glyphs
